src/nix/check.cc

```diff
@@ -0,0 +1,109 @@
+#include "nix/cmd/command.hh"
+#include "nix/cmd/installable-flake.hh"
+#include "nix/main/common-args.hh"
+#include "nix/main/shared.hh"
+#include "nix/store/store-api.hh"
+#include "nix/store/globals.hh"
+
+using namespace nix;
+
+struct CmdCheck : InstallablesCommand, MixDryRun
+{
+    std::string description() override
+    {
+        return "check that a derivation can be built or substituted";
+    }
+
+    std::string doc() override
+    {
+        return
+#include "check.md"
+            ;
+    }
+
+    Strings getDefaultFlakeAttrPaths() override
+    {
+        return {};
+    }
+
+    Strings getDefaultFlakeAttrPathPrefixes() override
+    {
+        return {
+            "checks." + settings.thisSystem.get() + ".",
+            "packages." + settings.thisSystem.get() + ".",
+            "legacyPackages." + settings.thisSystem.get() + "."};
```

Comment:
```suggestion
            "checks." + settings.thisSystem.get() + "."
            };
```

Having all the packages in the completions is counterproductive, and I haven't wanted to `check` buildability of a package yet.